docs/mkdocs/secureboot-build-and-sign.md

```diff
@@ -0,0 +1,86 @@
+# Signing kernel modules using KMM
+
+### Secure-boot with KMM
+
+For more details on using secure-boot see [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/managing_monitoring_and_updating_the_kernel/signing-kernel-modules-for-secure-boot_managing-monitoring-and-updating-the-kernel) or [here](https://wiki.debian.org/SecureBoot)
+
+### Using Signing with KMM
+
+On a secure-boot enabled system all kernel modules (kmods) must be signed with a public/private keypair enrolled into the Machine Owner's Key or MOK database. Drivers distributed as part of a distribution should already be signed by the distributions private key, but for kernel modules build out-of-tree KMM supports signing kernel modules using the "sign" section of the kernel mapping.
+
+KMM has the ability to build and sign kernel modules together by defining both the `build` and `sign` sections in the Module custom resource
+
+To use this functionality you need:
+
+ * A public private key pair in the correct (der) format
+ * At least one secure-boot enabled nodes with the public key enrolled in its MOK database
+ * The source code for the kmod to build and a Dockerfile for the driver container that will include it
+
+The yaml below should build a new container image using the source code from the https://github.com/chr15p/simple_driver_container (this kernel module does nothing useful but provides a good example). The image produced is saved back the the registry with a temporary name, and this temporary image is then signed using the parameters in the `sign` section.
+
+The temporary image name is based on the final image name (defined in `containerImage`) with `<namespace>_<module name>_kmm_unsigned` appended. So in the example below it would build an image named quay.io/chrisp262/minimal-driver:final-default_example-module_kmm_unsigned icontaining the build but unsigned kmods, and push it to the registry. Then it would create a second image, quay.io/chrisp262/minimal-driver:final containing the signed kmods. This second image that will then get loaded by the daemonset.
```

Comment:
```suggestion
The temporary image name is based on the final image name (defined in `containerImage`) with `<namespace>_<module name>_kmm_unsigned` appended. So in the example below it would build an image named quay.io/chrisp262/minimal-driver:final-default_example-module_kmm_unsigned containing the build but unsigned kmods, and push it to the registry. Then it would create a second image, quay.io/chrisp262/minimal-driver:final containing the signed kmods. This second image that will then get loaded by the daemonset.
```